kde/plasmoid/contents/scripts/read_file.py

```diff
@@ -9,16 +9,14 @@
   -h --help                     Show this screen.
   --debug                       Debug
 """
-from helper import config_effect_home, applet_effect_home, read_file
+from pathlib import Path
 from docopt import docopt
-import os.path
+from helper import effect_dirs, read_file
+
 arguments = docopt(__doc__)
 effect_name = arguments['<effect>']
 
-if effect_name.endswith(' '):
-    effect_name = effect_name[:-1]
-    effect_home = config_effect_home
-else:
-    effect_home = applet_effect_home
-s = read_file(os.path.join(effect_home, effect_name, arguments['<file>']))
-print(s)
+for effect_dir in effect_dirs:
+    effect = effect_dir / effect_name
+    if effect.is_dir():
+        print(read_file(effect / arguments['<file>']))
```

Comment:
Without "break", this line can read multiple files and print multiple times. Not a problem though, if I replace effect_name with a better format.

Reply:
right, maybe the best thing would be to create a central function that searches shaders, and that warns if a shader is found in two different directories.